### PR TITLE
feat: Use new JVS client to validate justification subject match

### PR DESCRIPTION
**Pull request by yolocs**

### CODE REVIEW

File: integration/testrunner/config.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ type Config struct {
 	LogRoutingWait          time.Duration `env:"AUDIT_CLIENT_TEST_AUDIT_LOG_ROUTING_WAIT,default=5s"`
 	MaxAuditLogRequestTries uint64        `env:"AUDIT_CLIENT_TEST_MAX_AUDIT_LOG_REQUEST_TRIES,default=4"`
 	MaxDBQueryTries         uint64        `env:"AUDIT_CLIENT_TEST_MAX_DB_QUERY_TRIES,default=60"`
+	ServivceAccount         string        `env:"SERVICE_ACCOUNT,required"`
```
Comment (sethvargo) [inline integration/testrunner/config.go:33]: `AUDIT_CLIENT_TEST_SERVICE_ACCOUNT`?

I could not find a suitable YouTube video.

File: integration/testrunner/grpc.go [hunk 2]
```diff
@@ -238,47 +234,6 @@ func testGRPCEndpoint(ctx context.Context, t *testing.T, g *GRPC) {
 	})
 }
 
-// create a justification token to pass in the call to services.
```
Comment (sethvargo) [inline integration/testrunner/grpc.go:241]: https://www.youtube.com/watch?v=Eo-KmOd3i7s

(no action required)

File: integration/testrunner/logging.go [hunk 3]
```diff
@@ -46,3 +53,44 @@ func MakeAuditLogRequest(id, endpointURL string, requestTimeout time.Duration, a
 	}
 	return resp, nil
 }
+
+// create a justification token to pass in the call to services.
+func justificationToken(audience, subject string) (string, error) {
```
Comment (sethvargo) [inline integration/testrunner/logging.go:58]: https://www.youtube.com/watch?v=u9Dg-g7t2l4

(no action required)

File: integration/testrunner/main_test.go [hunk 4]
```diff
@@ -68,8 +68,7 @@ func TestHTTPEndpoints(t *testing.T) {
 	}
 
 	for i, tc := range tests {
-		i := i
-		tc := tc
+		i, tc := i, tc
```
Comment (sethvargo) [inline integration/testrunner/main_test.go:71]: https://www.youtube.com/watch?v=O-zpOMYRi0w

(no action required)

File: integration/testrunner/config.go (hunk 1 above)
Comment (sethvargo) [inline integration/testrunner/config.go:33]: https://www.youtube.com/watch?v=0Wx-gt0jl4o

```suggestion
	ServiceAccount         string        `env:"SERVICE_ACCOUNT,required"`
```

Comment (sethvargo) [reply integration/testrunner/config.go:33]: But actually, I wonder if this should be more generic like "TestSubject" or something.

Comment (yolocs) [reply integration/testrunner/config.go:33]: Sure. Thinking `JustificationSubject`

Comment (yolocs) [reply integration/testrunner/config.go]: Done

Comment (yolocs) [reply integration/testrunner/config.go]: Done

**Review by sethvargo — APPROVED** [review]

**Pull request merged**